### ISSUE TITLE
Possibly expose exception text on error dialog

### ISSUE THREAD

**Issue by zadjii-msft** [microsoft/terminal#990]
Right now we display an error dialog if there was an exception parsing the settings.

We should potentially also add the exception test to that dialog, so the user can see exactly where the parsing went wrong. 

This is also dependent upon #475, where I'm adding support for `jsoncpp`, We can throw an exception with a decent message there, but we need to make sure it's actually displayed. 

Preferably, the exception text would
 1. Be copyable
 2. Be collapsible (so the user sees our message first, then can expand the "details")

**Comment by dlong11** [comment]
@zadjii-msft Wouldn't this cause localization issues?  Or are you thinking the exceptions strings will be resources that are translated?   Or do you think the details can always be English?

**Comment by zadjii-msft** [comment]
I think if we're displaying raw exception text, then that can just be in whatever locale the exception is originally. That would make searching for the exception text certainly easier

**Comment by DHowett-MSFT** [comment]
Stretch goal: add a copy button that puts the exc text on the clipboard ;P

**Comment by dlong11** [comment]
@zadjii-msft - I am not sure if I stated my question clearly.  

In the description you mention 
> We can throw an exception with a decent message there, but we need to make sure it's actually displayed.

My question is the "decent message" you mentioned above - Is this going to be translated?  Or will this be English?  This feature seems helpful but it also seems weird that if I am running under the German locale the exception text will be in English.  (if the plan is not to translate the exception messages)

I hope this makes sense.   English in these scenarios might OK, I am just trying to figure out what the plan is.

**Comment by zadjii-msft** [comment]
Ah see, I was thinking that error messages would have two sets of text.
1. The error message that we primarily display to the user
2. The raw exception text

1 would be localized, and would be text that we write ourselves. "Failed to load the settings file", etc. This text would always be displayed to the user. This is the text we have currently in the ContentDialog.

2 would not be localized, as it would be straight from whoever was throwing the exception. The case I'm thinking of would be exceptions thrown by jsoncpp, where it states the line and column where it failed to parse the file. This would only be displayed to the user when they chose to expand the exception text. Because it's not locale-specific, it would be easier to search the web for that _exact_ exception text, and find solutions.

I don't think we'd be capable of generically translating arbitrary exception messages from third-party libraries we might use.

With this model, we'd be able to have both a localized message the user can use to understand the message, and a specific message they can use to try and debug (or get help from us :P)

**Comment by dlong11** [comment]
@zadjii-msft Thanks!  This was what I was looking for.  😄   I was wondering how you were going to convert third party exceptions.   In the .NET world you are [expected to translate your exception](https://docs.microsoft.com/en-us/dotnet/standard/exceptions/best-practices-for-exceptions#include-a-localized-string-message-in-every-exception) messages.    In your scenario that would be a good thing, but in general, it always seemed a little weird.
